testsuite/gna/testsuite.sh

```diff
@@ -32,27 +32,29 @@ for opt; do
   esac
 done
 
-for i in $dirs; do
-  echo "GNA dir $i:"
-  cd $i
+singlerun() {
+  echo "GNA dir $1:"
+  cd $1
   if ! ./testsuite.sh; then
     echo "#################################################################"
-    echo "######### FAILURE: $i"
+    echo "######### FAILURE: $1"
     echo "#################################################################"
-    if [ $full = "y" ]; then
-      failures="$failures $i"
+    if [ $2 = "y" ]; then
+      failures="$failures $1"
     else
       exit 1;
     fi
   fi
   cd ..
+}
+
+for i in $dirs; do
+  printf "\033[34mGNA $i \033[0m\n" 1>> ../../log.log 2>&1
+  singlerun $i $full 1>> ../../log.log 2>&1
```

Comment:
Please, don't redirect in testsuite.sh (or add an option for that).
I often manually run then.

Not sure that colours are a good idea: AFAIK they don't work on Windows.
(An option is ok too).

(I understand progress is displayed on stdout and only test output is logged).